apps/web/scripts/remove-unused-translations.ts

```diff
@@ -0,0 +1,99 @@
+import fs from "fs";
+import glob from "glob";
+import path from "path";
+
+// All the paths are relative to the web app as the script will be run from there
+const ROOT_DIR = "../../"; // Directory to scan for used translation keys -> entire repository
+const TRANSLATIONS_DIR = "./public/static/locales"; // Directory containing translation files
+
+/**
+ * # Regex for translation keys
+ * This regex matches the following function calls and attribute:
+ * 1. t("<some_key>"),
+ * 2. t(`<some_key>`),
+ * 3. t("<some_key>", {key: "<some_string>"}),
+ * 4. t(`<some_key>`, {key: "<some_string>"}),
+ * 5. i18nKey="home_hero_subtitle",
+ * 6. i18nKey={`home_hero_subtitle`},
+ *
+ * It also ensure that we don't match any other similar function calls (e.g. `format("dddd")`).
+ **/
+const translationKeyRegex =
+  /(?<!\w)(?:(?:t|language)\((?:"|`)([^$"]*)(?:"|`)(?:,\s*\{[^}]*\})?\)|i18nKey=(?:{`|")([^$"]*)(?:`}|")[^\w])/g;
```

Comment:
Ideally this would be perfect, but these patterns aren't always being followed. There are keys that are fall outside these patterns and gets removed.


https://github.com/user-attachments/assets/20b0520d-c3c2-494c-a9be-bb3e85a24220



![Screenshot 2024-08-29 194110](https://github.com/user-attachments/assets/b984b438-5768-408f-8428-7adb9030be7b)

![Screenshot 2024-08-29 194128](https://github.com/user-attachments/assets/c54699c6-c046-4fed-a44c-9c2b809dda5c)

Reply:
I think we should just check it against every string instead of only some specific patterns. Performance might take a hit, but since we won't be running this very often, it would be an acceptable trade-off.

Reply:
Yeah, I was not aware of this use case in the trpc router. 

However, checking all the keys against files is also not viable because the keys are also dynamically generated in code like this `booking_${needsConfirmation ? "submitted" : "confirmed"}${recurringBookings ? "_recurring" : ""}`.